src/runtime-rs/crates/agent/src/kata/mod.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+mod agent;
+mod trans;
+
+use std::os::unix::io::{IntoRawFd, RawFd};
+
+use anyhow::{Context, Result};
+use kata_types::config::Agent as AgentConfig;
+use protocols::{agent_ttrpc_async as agent_ttrpc, health_ttrpc_async as health_ttrpc};
+use tokio::sync::Mutex;
+use ttrpc::asynchronous::Client;
+
+use crate::{
+    log_forwarder::LogForwarder,
+    vsock::{connect_vsock, VsockConnectConfig},
+};
+
+// https://github.com/firecracker-microvm/firecracker/blob/master/docs/vsock.md
+#[derive(Debug, Default)]
+pub struct Vsock {
+    pub context_id: u64,
+    pub port: u32,
+}
+
+pub(crate) struct KataAgentInner {
+    pub client: Option<Client>,
+    pub client_fd: RawFd,
+    pub uds: String,
+
+    /// Agent server port
+    pub server_port: u32,
```

Comment:
These fields are the same as `Agent` struct in `src/libs/kata-types/src/config/agent.rs`, so can save `Agent` as a field in `KataAgentInner`?

Reply:
Thanks for your review, change to use AgentConfig as a field in KataAgentInner.